apis/duck/unstructured_test.go

```diff
@@ -70,23 +73,87 @@ func TestFromUnstructuredFooable(t *testing.T) {
 		want:      FooStatus{},
 		wantError: nil,
 	}}
+
 	for _, tc := range tcs {
-		raw, err := json.Marshal(tc.in)
-		if err != nil {
-			panic("failed to marshal")
-		}
+		t.Run(tc.name, func(t *testing.T) {
+			raw, err := json.Marshal(tc.in)
+			if err != nil {
+				panic("failed to marshal")
+			}
+
+			t.Logf("Marshalled : %s", string(raw))
+
+			got := Foo{}
+			err = FromUnstructured(tc.in, &got)
+			if err != tc.wantError {
+				t.Fatalf("Unexpected error for %q: %v", string(tc.name), err)
+			}
+
+			if !reflect.DeepEqual(tc.want, got.Status) {
```

Comment:
`got, want := ...`, cmp.Equal to compare, cmp.Diff in the `Errorf` call.

Reply:
Done